pubSubHandlers/banHandler.py

```diff
@@ -16,12 +16,22 @@ async def handle(self, raw_data: bytes) -> None:
         await user_utils.remove_from_leaderboard(userID)
         await user_utils.remove_first_place(userID)
 
-        if not (targetToken := await osuToken.get_token_by_user_id(userID)):
-            return
-
-        targetToken["privileges"] = await user_utils.get_privileges(userID)
-        await osuToken.disconnectUserIfBanned(targetToken["token_id"])
-        await osuToken.checkRestricted(targetToken["token_id"])
+        new_privileges = await user_utils.get_privileges(userID)
+
+        all_user_tokens = await osuToken.get_all_tokens_by_user_id(userID)
+
+        for token in all_user_tokens:
+            maybe_token = await osuToken.update_token(
+                token["token_id"],
+                privileges=new_privileges,
+            )
+            assert maybe_token is not None
+            token = maybe_token
```

Comment:
?

Reply:
`token` has type `Token`, `maybe_token` has type `Token | None`, assigning `Token | None` to `Token` is something mypy recommends against, so i go with a temp name here. Maybe could have a pattern like `update_token(token_id, existence_check=True)` which overloads the return type to `Token`

Reply:
I think what confused me most was choosing `token` as naming when the loop is using `token` too. Just reads incredibly weird and couldn't really figure out what was going on here with the assert and everything...

Reply:
Hmm at the moment it would be fine to ignore the return type, but I guess more philosophically I don't like leaving references to things which have been updated in redis but not in memory